src/NuGet.Core/NuGet.Common/PathUtil/PathUtility.cs

```diff
@@ -94,6 +100,27 @@ public static bool IsChildOfDirectory(string dir, string candidate)
             candidate = Path.GetFullPath(candidate);
             return candidate.StartsWith(dir, StringComparison.OrdinalIgnoreCase);
         }
+
+        public static bool HasTrailingValidDirectorySeparator(string path)
+        {
+            if (string.IsNullOrEmpty(path))
+            {
+                return false;
+            }
+            else
+            {
+                if (RuntimeEnvironmentHelper.IsWindows)
```

Comment:
Seems like this check isn't needed. It would just skip the alt char check but it should be fine to check it also I would think.